### PR TITLE
Update svgo to v2.4.0

### PR DESCRIPTION
**Pull request by XhmikosR**

### CODE REVIEW

**Comment by XhmikosR** [comment]
@TrySound: it seems I messed this up. Could you please have a look and suggest the best way moving forward?

PS. if you need push rights let me know.

File: example.js [hunk 1]
```diff
@@ -11,7 +11,34 @@ const files = glob.sync('**/weather*.svg', { cwd });
...
+    log: 'debug',
+    shape: {
+        transform: [{
+            svgo: {
+                multipass: true,
+                plugins: [{
+                    name: 'preset-default',
+                    params: {
+                        overrides: {
+                            cleanupListOfValues: true,
```
Comment (TrySound) [inline example.js:23]: cleanupListOfValues is not part of default preset. It's one of builtin plugins which can be added before or after preset-default plugin. Same with other not default plugins.

Comment (XhmikosR) [reply example.js:23]: I'm not sure how to proceed. Note that this is an example, the code is in lib/svg-sprite/transform/svgo.js.

I tried to use the master branch on a project and I couldn't get things to work, that's why I asked for help :)

Comment (TrySound) [reply example.js:23]: What doesn't work?

Comment (XhmikosR) [reply example.js:23]: This is my test branch: https://github.com/twbs/icons/compare/update-svg-sprite

I notice that `xmlns` are still added. Maybe it's just a stupid thing of wrong merging of the options in this PR...

Comment (TrySound) [reply example.js:23]: Same removeXMLNS is not part of preset-default

Comment (TrySound) [reply example.js:23]: I should probably warn about invalid plugins

Comment (XhmikosR) [reply example.js:23]: Yeah, agreed. A warning would definitely help :)

So, does this mean that one cannot use this plugin anymore? If it's possible, how can one do it? I find the documentation a little unclear.

Thanks for the help!

PS. I hope I'll manage to ship v2.0.0 of svg-sprite which has svgo 2.x, worst case scenario I revert this PR and we'll get the warnings until someone fixes it properly.

Comment (XhmikosR) [reply example.js:23]: OK, I read again the docs and it's clear how to include it. All good, after all!

I will revisit this PR to make sure all options work.

Comment (TrySound) [reply example.js:23]:
```
[
  {
    name: 'preset-default',
    params: {
      overrides: {
        removeViewBox: false
      }
    }
  },
  'removeXMLNS'
]
```

Comment (XhmikosR) [reply example.js:23]: I still can't make `sortAttrs` work with this config:

```json
{
  "mode": {
    "symbol": {
      "dest": ".",
      "sprite": "bootstrap-icons.svg"
    }
  },
  "shape": {
    "transform": [
      {
        "svgo": {
          "multipass": true,
          "plugins": [
            {
              "name": "preset-default",
              "params": {
                "overrides": {
                  "removeAttrs": {
                    "attrs": [
                      "data-name",
                      "fill",
                      "clip-rule"
                    ]
                  },
                  "removeUnknownsAndDefaults": {
                    "keepRoleAttr": true
                  },
                  "removeViewBox": false
                }
              }
            },
            "cleanupListOfValues",
            "removeXMLNS",
            "sortAttrs"
          ]
        }
      }
    ]
  },
  "svg": {
    "namespaceClassnames": false,
    "xmlDeclaration": false
  }
}
```

I should probably move `removeAttrs` outside of `default-plugin` but it seems to work even if it's inside like above.

**Pull request merged**